README.md

```diff
@@ -1 +1,139 @@
-# k8s-ipsec-cni
\ No newline at end of file
+# k8s-ipsec-cni
+
+Sample implementation of a 3-node Kubernetes cluster using strongSwan and a host-local CNI plugin to provide IPSEC-based networking for pods.
+
+The CNI plugin configuration is mostly taken from the [CNI plugin reference](https://github.com/containernetworking/cni/blob/master/SPEC.md
+).
+
+Requirements:
+* Working installation of Hashicorp Vagrant 2 (https://www.vagrantup.com/) with either of these provisioners:
+    * virtualbox
+    * libvirt
+
+### Deployment
+
+1. Check out this repo into a directory: ```git clone https://github.com/webvictim/k8s-ipsec-cni```
+
+2. cd into the directory: ```cd k8s-ipsec-cni```
+
+3. Run ```vagrant up --provision```
+
+The initial startup/provisioning process takes around 10-15 minutes depending on the speed of the test machine and the internet connection.
+
+Once the provisioning process is complete, SSH into the Kubernetes master with ```vagrant ssh master```.
+
+We should then be able to see three working nodes:
+
+```
+vagrant@kube-master:~$ kubectl get nodes
+NAME          STATUS     ROLES     AGE       VERSION
+kube-master   NotReady   master    8m        v1.10.0
```

Comment:
In the exercise specification it was not specified to use a different node as a master.
We're interested in knowing your ideas in going down this path.